### PR TITLE
pkg/columns: add support for embedded structs via pointer

### PR DESCRIPTION
**Pull request by flyth**
This PR adds the ability to embed structs non-anonymously and have them handled by the columns library.
This idea/requirement for this came up [here](https://github.com/inspektor-gadget/inspektor-gadget/pull/1167#discussion_r1041030558).

Example from the tests:
```golang
type embeddedStruct struct {
	EmbeddedString string `column:"embeddedString"`
}
type embeddedPtrStruct struct {
	EmbeddedString2 string `column:"embeddedString2"`
}
type ptrStruct struct {
	EmbeddedString string `column:"ptrStructString"`
}
type normalStruct struct {
	EmbeddedString string `column:"normalStructString"`
}
type testStruct struct {
	embeddedStruct
	*embeddedPtrStruct
	PointerStruct           *ptrStruct
	NormalStruct            normalStruct
	NotEmbeddedPtrStruct    *ptrStruct   `column:"ptrStruct,noembed"`
	NotEmbeddedNormalStruct normalStruct `column:"normalStruct,noembed"`
}
```

Anonymous structs (`embeddedStruct` and `embeddedPtrStruct` here) will automatically be handled by the columns library. New is the handling of anonymous structs by pointers (`embeddedPtrStruct`).

Named structs will now be handled as if they were embedded unless they're having a `noembed` columns-tag. All members of the embedded struct will then become available using their column names (if a column tag is set). If the `noembed` tag is set, the struct (or pointer to struct) will be handled like previously (and by default extracted using fmt.Sprintf("%v")).

In case pointers to structs are nil, the returned values for the columns will equal the zero value of that type (e.g. empty string for string, 0 for ints and so on.)

#### Additional features we can add later on (if needed)

If we need to, we could add prefixes, so that for example
```golang
type testStruct struct {
	PointerStruct           *ptrStruct   `column:",prefix:ptr."`
}
```
will lead to a column named `ptr.ptrStructString`. This would be needed if the same struct is embedded more than once (otherwise the column names would clash.)

### CODE REVIEW

**Review by mqasimsarfraz — APPROVED** [review]
LGTM! I tested these changes are are working fine, thanks for handling it so quickly!

File: pkg/columns/columns.go [hunk 1]
```diff
@@ -151,11 +151,19 @@ func (c ColumnMap[T]) VerifyColumnNames(columnNames []string) (valid []string, i
 	return
 }
 
-func (c *Columns[T]) iterateFields(t reflect.Type, sub []int, offset uintptr) error {
+func (c *Columns[T]) iterateFields(t reflect.Type, sub []subField, offset uintptr) error {
+	isPtr := false
+	if t.Kind() == reflect.Pointer {
+		if t.Elem().Kind() != reflect.Struct {
+			return errors.New("unsupported pointer type")
```
[the remark this answers is not in the document]
Comment (flyth) [reply pkg/columns/columns.go:158]: Good question: right now that would only be hit with something like that:
```golang
type B string

type A struct {
	*B `column:"foo"`
}
```
And that should be a valid thing to do, actually.

So I'll also add that check to the anonymous fields to make sure these pointers to non-structs will also be handled as normal fields. Regarding the actual check you mentioned: I would still leave that there to make sure the input is valid when changing the lib later on. This is no hot code (called once per type), so I think more checks to clarify what is expected also help understanding the library better.

**Comment by flyth** [comment]
Thanks for the review, @mqasimsarfraz!

I'm still not entirely happy with handling embedded anonymous structs automatically, while named structs are required to have the `embed` tag. Maybe we should reverse that and "embed" automatically (to be consistent with current behavior with the anonymous structs) and require a `noembed` tag to handle the field as a single column. WDYT?

**Comment by burak-ok** [comment]
>  "embed" automatically (to be consistent with current behavior with the anonymous structs) and require a `noembed` tag to handle the field as a single column.

If we do not "embed" automatically it is consistent with the behavior of other namend variables like `myInt int` etc. One line equals one column. I think this is just a definition we need to set somewhere and document it.

Nonetheless, I would also like to have them embedded automatically. Higher usecase

**Comment by mqasimsarfraz** [comment]
> I'm still not entirely happy with handling embedded anonymous structs automatically, while named structs are required to have the `embed` tag. Maybe we should reverse that and "embed" automatically (to be consistent with current behavior with the anonymous structs) and require a `noembed` tag to handle the field as a single column. WDYT?

Agreed. I was thinking `go` itself treats field access for embedded and named nested structs differently but in context of `column` tag I would expect `embed` to be default like you proposed.

**Review by burak-ok — APPROVED** [review]
LGTM

**Pull request merged**